test/unit/org/apache/cassandra/net/SensorsCustomParamsTest.java

```diff
@@ -95,4 +118,72 @@ public void testIndexReadBytesTableParam()
+        TableMetadata tm = TableMetadata.builder("ks1", "t1", TableId.fromString(tableId.toString()))
+                                        .addPartitionKeyColumn("pk", AsciiType.instance)
+                                        .build();
+        SensorsRegistry.instance.onCreateKeyspace(ksm);
+        SensorsRegistry.instance.onCreateTable(tm);
+
+        Context context = new Context("ks1", "t1", tableId.toString());
+        sensors.registerSensor(context, Type.WRITE_BYTES);
+        sensors.incrementSensor(context, Type.WRITE_BYTES, 17.0);
+        sensors.syncAllSensors();
+
+        Message.Builder<NoPayload> builder =
+        Message.builder(Verb._TEST_1, noPayload)
+               .withId(1);
+
+        SensorsCustomParams.addWriteSensorToResponse(builder, sensors, context);
+
+        Message<NoPayload> msg = builder.build();
+        assertNotNull(msg.header.customParams());
+        assertEquals(2, msg.header.customParams().size());
+        String requestParam = SensorsCustomParams.encodeTableInWriteBytesRequestParam("t1");
+        String tableParam = SensorsCustomParams.encodeTableInWriteBytesTableParam("t1");
+        assertTrue(msg.header.customParams().containsKey(requestParam));
+        assertTrue(msg.header.customParams().containsKey(tableParam));
+        double epsilon = 0.000001;
+        assertEquals(17.0, SensorsCustomParams.sensorValueFromBytes(msg.header.customParams().get(requestParam)), epsilon);
+        assertEquals(17.0, SensorsCustomParams.sensorValueFromBytes(msg.header.customParams().get(tableParam)), epsilon);
+    }
+
+    @Test
+    public void testAddReadSensorToResponse()
+    {
+        RequestSensors sensors = new RequestSensors();
+        UUID tableId = UUID.randomUUID();
+        KeyspaceMetadata ksm = KeyspaceMetadata.create("ks1", null);
+        TableMetadata tm = TableMetadata.builder("ks1", "t1", TableId.fromString(tableId.toString()))
+                                        .addPartitionKeyColumn("pk", AsciiType.instance)
+                                        .build();
+        SensorsRegistry.instance.onCreateKeyspace(ksm);
+        SensorsRegistry.instance.onCreateTable(tm);
+
+        Context context = new Context("ks1", "t1", tableId.toString());
+        sensors.registerSensor(context, Type.READ_BYTES);
+        sensors.incrementSensor(context, Type.READ_BYTES, 13.0);
+        sensors.syncAllSensors();
+
+        Message.Builder<NoPayload> builder =
+        Message.builder(Verb._TEST_1, noPayload)
+               .withId(1);
+
+        SensorsCustomParams.addReadSensorToResponse(builder, sensors, context);
+
+        Message<NoPayload> msg = builder.build();
+        assertNotNull(msg.header.customParams());
+        assertEquals(2, msg.header.customParams().size());
+        assertTrue(msg.header.customParams().containsKey(SensorsCustomParams.READ_BYTES_REQUEST));
+        assertTrue(msg.header.customParams().containsKey(SensorsCustomParams.READ_BYTES_REQUEST));
```

Comment:
I guess this one should be `READ_BYTES_TABLE`

Reply:
Yes fixed indirectly by the refactoring you suggested